### PR TITLE
DBTP-364 IP filter validation etc

### PR DESCRIPTION
**Pull request by antroy-madetech**
From the ticket:

Schema validation of AppConfig data should be done in the IP filter so that broken config is rejected with a log message rather than crashing the IP filter.  

Also we need tests to verify the IP filter gracefully handles: 

404 from the AppConfig agent
An error from the AC agent due to policy/access issues
A missing AppConfig agent

For security, I have made all the appconfig related errors cause the page to be blocked.

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -2,7 +2,4 @@
 parallel = True
 branch = True
 concurrency = multiprocessing
-include =
-	main.py
-	tests.py
-	utils.py
+source = .
```
Comment (acodeninja) [inline .coveragerc:5]: EOF newline.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+[tool.poetry]
+name = "ip-filter"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
```
Comment (acodeninja) [inline pyproject.toml:5]: Maybe the webops team?

File: README.md [hunk 3]
```diff
@@ -122,3 +122,39 @@ SERVER: localhost:8080
 APPCONFIG_PROFILES: default:rule:set
 ```
 
+## Contributing to ip-filter
+
+### Getting started
+
+1. Clone the repository:
+
+   ```
```
Comment (acodeninja) [inline README.md:131]:
```suggestion
   ```shell
```

File: README.md [hunk 4]
```diff
@@ -122,3 +122,39 @@ SERVER: localhost:8080
...
+   git clone https://github.com/uktrade/ip-filter.git && cd ip-filter
+   ```
+
+2. Install the required dependencies:
+
+   ```
```
Comment (acodeninja) [inline README.md:137]:
```suggestion
   ```shell
```

**Review by acodeninja — COMMENTED** [review]
Few small changes.

**Review by acodeninja — APPROVED** [review]
Happy mine are covered, assuming @WillGibson approves based on his suggestions, so do I.

**Review by WillGibson — APPROVED** [review]

**Pull request merged**